vars/artifactSetVersion.groovy

```diff
@@ -52,12 +52,9 @@ def call(Map parameters = [:], Closure body = null) {
             .dependingOn('buildTool').mixin('filePath')
             .dependingOn('buildTool').mixin('versioningTemplate')
             .use()
-
-        config = new ConfigurationHelper(config)
-            .addIfEmpty('gitSshUrl', (config.buildTool == 'docker' && config.artifactType == 'appContainer')?script.commonPipelineEnvironment.getAppContainerProperty('gitSshUrl'):script.commonPipelineEnvironment.getGitSshUrl())
-            .addIfEmpty('timestamp', getTimestamp(config.timestampTemplate))
-            .use()
-
+        
+        config.timestamp = config.timestamp ?: getTimestamp(config.timestampTemplate))
```

Comment:
IMO we should abandon the pattern of changing the configuration outside the config helper. IMO we should return a immutable map (of course deep!!!) from the configuration framework. Having coding which updates the config map directly makes it harder to switch to immutables later.

Why not add the ```.addIfEmpty```to the config helper instanciated in line 44?

Reply:
we need to reference the config itself

Reply:
... aaaahh, I see.
I this case I vote for opening opening a new ```ConfigHelper``` instance rather than manipulating the config directly.